### DISCUSSION TITLE
Support for other storage backends

### DISCUSSION THREAD

**Discussion by mattpodraza** [einride/iam-go#104] [Ideas]
First off - this looks really interesting, I'm impressed :)

Are there any plans to support other storage backends or will it only ever be Cloud Spanner?

**Comment by odsod** [comment]
Hi @mattpodraza! Thanks!

Right now we're still using Spanner as a POC for the whole thing - interfaces and APIs will probably change a few times more over the summer. But the external API of the `iamspanner` package is meant to be usable as a generic interface for other storage backends in the future.

What database would you be interested in using this with?

**Reply by mattpodraza** [reply]
I was thinking that CockroachDB would be a possibility, as it's meant to be an open source alternative to Spanner.

Or, for smaller-scale operations, PostgreSQL would probably suffice.

The one problem I have with Spanner is that it's quite on the pricey side.
